failing cftime plot tests

This was reported in #5077 and seems to be an upstream issue (see https://github.com/pydata/xarray/issues/5077#issuecomment-808605313):

```
FAILED xarray/tests/test_plot.py::TestCFDatetimePlot::test_cfdatetime_line_plot
FAILED xarray/tests/test_plot.py::TestCFDatetimePlot::test_cfdatetime_pcolormesh_plot
FAILED xarray/tests/test_plot.py::TestCFDatetimePlot::test_cfdatetime_contour_plot
```

In order to get the upsteam-dev CI open new issues for other failing tests on upstream-dev I've `xfail`'ed them in #5343. This should be undone once it is fixed in either `cftime` or `nc-time-axis`.

--- a/xarray/tests/test_plot.py
+++ b/xarray/tests/test_plot.py
@@ -2581,7 +2581,6 @@ class TestDatetimePlot(PlotTestCase):
         self.darray.plot.line()
 
 
-@pytest.mark.xfail(reason="recent versions of nc-time-axis and cftime are incompatible")
 @pytest.mark.filterwarnings("ignore:setting an array element with a sequence")
 @requires_nc_time_axis
 @requires_cftime

